trinity/main.py

```diff
@@ -65,27 +73,85 @@ def main():
     db_path = get_data_dir(ROPSTEN)
     os.makedirs(db_path, exist_ok=True)
 
+    db_process_pipe, db_connection_pipe = ctx.Pipe()
+
     # For now we just run the light sync against ropsten by default.
-    process = ctx.Process(
+    chain_process = ctx.Process(
         target=ropsten_light_node_sync,
-        args=(log_queue, db_path),
+        args=(log_queue, db_connection_pipe),
+    )
+    db_process = ctx.Process(
+        target=base_db_process,
+        args=(LevelDB, (db_path,), db_process_pipe, log_queue),
     )
 
     try:
-        process.start()
-        process.join()
+        db_process.start()
+        chain_process.start()
+        chain_process.join()
     except KeyboardInterrupt:
         logger.info('Keyboard Interrupt: Stopping')
-        process.terminate()
+        chain_process.terminate()
+        db_process.terminate()
 
 
-def ropsten_light_node_sync(log_queue, db_path):
+def base_db_process(db_class, init_args, mp_pipe, log_queue):
```

Comment:
I expect it to be a pretty large refactor to handle multiple pipes and async, so this is a note to self that I'd be curious to see how that plays out. When I did the IPC server, I "cheated" and used a built-in server, but I don't think that has any explicit multiprocess support. For reference: https://github.com/ethereum/py-evm/blob/master/evm/rpc/ipc.py#L99-L104

Reply:
I *think* that wiring this in will be as simple as modifying the RPCServer to use the `PipeDB` (previously `MPDB`) class for it's backend db connection.